### PR TITLE
Extend `sql-db/hibernate-reactive` and `sql-db/vertx-sql` with `oracle`

### PR DESCRIPTION
**Pull request by jsmrcka**
Extend with reactive Oracle client.
Implementation of test plan for QUARKUS-1407:
https://github.com/quarkus-qe/quarkus-test-plans/blob/main/QUARKUS-1407.md

Related issues/PRs:
- ~~https://github.com/quarkus-qe/quarkus-test-framework/pull/428 - Reactive URL for OracleService not yet implemented in TF.~~ Current implementation is outdated. New issues filed:
  - Upstream: https://github.com/quarkusio/quarkus/issues/26426 - Clarify correct reactive URI syntax in documentation.
  - TF: https://github.com/quarkus-qe/quarkus-test-framework/issues/503 - Fix the reactive URI in OracleService and add tests.
- https://issues.redhat.com/browse/QUARKUS-1164?focusedCommentId=18977100&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-18977100 - Oracle image does not work in OpenShift.
- https://github.com/quarkusio/quarkus/issues/24264 - No Oracle-specific exception available.
- ~~https://github.com/quarkus-qe/quarkus-test-suite/issues/566 - Unresolved issue of returning last inserted ID from Oracle.~~
- `OracleDatabaseIT#ensureSessionIsPropagatedOnReactiveTransactions` - Test disabled, issue created: https://github.com/quarkusio/quarkus/issues/24501
- ~~`sql-db/vertx-sql` - native build disabled: https://github.com/quarkusio/quarkus/issues/24788~~ Resolved

### CODE REVIEW

File: sql-db/vertx-sql/pom.xml [hunk 1]
```diff
@@ -79,6 +87,20 @@
         </dependency>
     </dependencies>
     <profiles>
+        <profile>
```
Comment (fedinskiy) [inline sql-db/vertx-sql/pom.xml:90]: I don't like the idea of disabling native for all tests. Can we do the opposite, and include oracle dependencies only when native is disabled?

**Review by fedinskiy — CHANGES_REQUESTED** [review]

File: sql-db/vertx-sql/pom.xml (hunk 1 above)
Comment (jsmrcka) [reply sql-db/vertx-sql/pom.xml:90]: Not really, I need Oracle dependencies in the code. What I can do is remove oracle from `vertx-sql` and create a new module `vertx-sql-oracle`, which would be analogous, but dedicated just to Oracle, and have native disabled. And we could merge those two modules once the issue with native build is resolved, similarily to what was done in https://github.com/quarkus-qe/quarkus-test-suite/pull/434.

File: sql-db/hibernate-reactive/src/test/java/io/quarkus/ts/reactive/OracleDatabaseIT.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import io.quarkus.test.services.Container;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class OracleDatabaseIT extends AbstractReactiveDatabaseIT {
+
+    private static final String ORACLE_USER = "quarkus_test";
+    private static final String ORACLE_PASSWORD = "quarkus_test";
+    private static final String ORACLE_DATABASE = "quarkus_test";
+    static final int ORACLE_PORT = 1521;
```
[the remark this answers is not in the document]
Comment (jsmrcka) [reply sql-db/hibernate-reactive/src/test/java/io/quarkus/ts/reactive/OracleDatabaseIT.java:18]: No, it should be private. Thx for noticing.

File: sql-db/vertx-sql/pom.xml (hunk 1 above)
Comment (jsmrcka) [reply sql-db/vertx-sql/pom.xml:90]: @rsvoboda WDYT?
I see 3 options:

1. Drop Oracle form `vertx-sql` and create a clone module just for Oracle.
2. Drop Oracle from `vertx-sql` with no substitution.
3. Disable native in `vertx-sql`.

Comment (jsmrcka) [reply sql-db/vertx-sql/pom.xml]: The native issue has been resolved (sice 2.10.0.Final). I think we can proceed with this PR. The remaining unresolved issues are listed in the PR description.

**Review by rsvoboda — APPROVED** [review]

File: sql-db/vertx-sql/src/test/java/io/quarkus/ts/vertx/sql/dbpool/PostgresPoolTest.java [hunk 3]
```diff
@@ -27,6 +28,7 @@
 import io.vertx.mutiny.pgclient.PgPool;
 import io.vertx.mutiny.sqlclient.RowSet;
 
+@Disabled("TMP - remove me!!!")
```
Comment (fedinskiy) [inline sql-db/vertx-sql/src/test/java/io/quarkus/ts/vertx/sql/dbpool/PostgresPoolTest.java:31]: I guess, it should be removed :)

File: sql-db/hibernate-reactive/src/test/java/io/quarkus/ts/hibernate/reactive/OracleDatabaseIT.java [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+import io.quarkus.test.services.Container;
+import io.quarkus.test.services.QuarkusApplication;
+
+@QuarkusScenario
+public class OracleDatabaseIT extends AbstractDatabaseHibernateReactiveIT {
+
+    private static final String ORACLE_USER = "quarkus_test";
+    private static final String ORACLE_PASSWORD = "quarkus_test";
+    private static final String ORACLE_DATABASE = "quarkus_test";
+    static final int ORACLE_PORT = 1521;
```
Comment (fedinskiy) [inline sql-db/hibernate-reactive/src/test/java/io/quarkus/ts/hibernate/reactive/OracleDatabaseIT.java:18]: This variable is suspiciously not private. Is it intentional?

File: sql-db/vertx-sql/src/main/java/io/quarkus/ts/vertx/sql/services/DbPoolService.java [hunk 5]
```diff
@@ -4,13 +4,19 @@
...
+import io.vertx.mutiny.oracleclient.OracleClient;
 import io.vertx.mutiny.sqlclient.Pool;
 import io.vertx.mutiny.sqlclient.SqlClientHelper;
+import io.vertx.oracleclient.OraclePrepareOptions;
 import io.vertx.sqlclient.PropertyKind;
 
 public class DbPoolService extends Pool {
 
     private final static PropertyKind<Long> LAST_INSERTED_ID = PropertyKind.create("last-inserted-id", Long.class);
+    public static final String ORACLE_AUTO_GENERATED_KEY_NAME = "ID";
```
Comment (fedinskiy) [inline sql-db/vertx-sql/src/main/java/io/quarkus/ts/vertx/sql/services/DbPoolService.java:17]: Why should this and the next variable be the `public` ones?

**Review by fedinskiy — CHANGES_REQUESTED** [review]

File: sql-db/hibernate-reactive/src/test/java/io/quarkus/ts/hibernate/reactive/OracleDatabaseIT.java (hunk 4 above)
Comment (jsmrcka) [reply sql-db/hibernate-reactive/src/test/java/io/quarkus/ts/hibernate/reactive/OracleDatabaseIT.java:18]: You're right, this should be private. I shuffled with old commits and re-introduced a mistake that has already been fixed before.

File: sql-db/vertx-sql/src/main/java/io/quarkus/ts/vertx/sql/services/DbPoolService.java (hunk 5 above)
Comment (jsmrcka) [reply sql-db/vertx-sql/src/main/java/io/quarkus/ts/vertx/sql/services/DbPoolService.java:17]: They should be private, thx.

**Pull request merged**